[jdbc-v2] bug fix for UInt32 UInt64 and Added support for Nested Type

gfunc opened the pull request:
## Summary
<!-- A short description of the changes with a link to an open issue. -->

1.  fixed bug for ResultSet.getObject() for type UInt32 UInt64
2. added support for not flattened Nested type in BinaryStreamReader
3. ~~added support for  set session settings and set profile~~

Closes <!-- Link to an issue to close on merge. -->
## Checklist
Delete items not relevant to your PR:
- [x] Closes #2333
- [x] Unit and integration tests covering the common scenarios were added

gfunc commented:
Fixing test errors

gfunc commented:
create table with setting `flatten_nested` is not supported in ClickHouse version 24.3 and 24.8. Thus the tests for non-flatten Nested type reported error.

gfunc commented:
> It is worth mentioning that I change the behavior of `USE` statement as well. I passed the database assigned by `USE` statement to connection rather than only keeping it within Statement. Since all the session settings are stored in connection, and to my knowledge `USE` statement should take effect session wide, it would be reasonable to pass the updated database to connection and treat it as the new default.

correction: previous `USE` statement should be broken because of this code in `executeQuery()`. 
https://github.com/ClickHouse/clickhouse-java/blob/5ae9b837d546257e5edaaac149bec7267d3a621f/jdbc-v2/src/main/java/com/clickhouse/jdbc/StatementImpl.java#L187

already fixed and added test to cover `executeQuery`

chernser commented on jdbc-v2/src/main/java/com/clickhouse/jdbc/StatementImpl.java:
```diff
@@ -376,14 +376,28 @@ public boolean execute(String sql, QuerySettings settings) throws SQLException {
                     }
                 }
+
+            } else if (JdbcUtils.containsIgnoresCase(tokens, "=")){
```
this will be true for any string containing "=" 
it may be would work for simple case. 
just to note.

chernser commented on jdbc-v2/src/main/java/com/clickhouse/jdbc/StatementImpl.java:
```diff
@@ -376,14 +376,28 @@ public boolean execute(String sql, QuerySettings settings) throws SQLException {
+                String key = tokens.get(1);
+                String value = tokens.get(3).replace("'", "");
+                //SET profile
+                if (key.equals("profile")) {
```
`equalsIgnoreCase` ?

gfunc replied:
Settings and profile seem to be case sensitive, and ClickHouse will return error `UNKNOWN_SETTING` if the key is not recognized

chernser commented:
Good day, @gfunc !

Would you please explain your fix for the UInt32/UInt64? 
We are going to change statement parsing - would you please just check server version in tests to skip them? 

Thanks you!

gfunc commented:
Good day! @chernser 

Since the recent releases changed how the v2 driver chooses JDBC type and Java class for ClickHouse data types, it seems it will force the same Java class for the same JDBC type. 

My fix for UInt32/Uint64 is simple: 

- return JDBC Type `BigInt` and Java type Long for `UInt32`, therefore skipped the conversion from Long defined `BinaryStreamReader` to Integer in function `JdbcUtils.convert`
- remove `UInt64` from `CLICKHOUSE_TO_SQL_TYPE_MAP` in `JdbcUtils` and, like `Int128`,  JDBC Type `Other` and Java Type `BigInteger` defined in `BinaryStreamReader` will be returned directly

As for changes related to the `SET` statement, I will try to skip the test for non-flattened nested type for ClickHouses version 24.3 and 24.8 for now and maybe raise another PR later.

chernser commented on jdbc-v2/src/main/java/com/clickhouse/jdbc/internal/JdbcUtils.java:
```diff
@@ -44,9 +44,8 @@ private static Map<ClickHouseDataType, SQLType> generateTypeMap() {
-        map.put(ClickHouseDataType.UInt32, JDBCType.INTEGER);
+        map.put(ClickHouseDataType.UInt32, JDBCType.BIGINT);
         map.put(ClickHouseDataType.Int64, JDBCType.BIGINT);
-        map.put(ClickHouseDataType.UInt64, JDBCType.BIGINT);
```
This means that UInt64 is not mapped to anything, right? I think it is not correct. 
UInt64 is not Other just because it is represent by BigInteger. This mapping is mainly for metadata.

chernser replied:
You are right in some way. 
But UInt32 and UInt64 are both OTHER because JDBC type is valid only for signed numbers. 
Would you please remove UInt8, UInt16, UInt32 and UInt64 from this mapping? We should return whatever BinaryStreamReader returns. 

Thanks!

gfunc replied:
all Uint had been removed

chernser commented:
@gfunc 
Thank you for the explanation! 
But we have this mapping https://clickhouse.com/docs/sql-reference/data-types/int-uint#integer-aliases and we have to keep UInt64 as BIGINT otherwise it will screw type matching. 
Give me sometime to think how is it better to handle. Definitely UInt32 in java is Long and UInt64 is BigInteger (as BinaryStreamReader returns)

chernser commented:
@gfunc 
Please see my comments. Thank you for pointing out the problem!

gfunc commented:
Hi @chernser 
I had removed `SET` related code in this PR and removed all Uint in `CLICKHOUSE_TO_SQL_TYPE_MAP` .
And workflow ran with only errors relateed to create database error for CH cloud.
